pkg/apis/provisioning/v1alpha1/provisioner.go

```diff
@@ -137,27 +137,20 @@ type ProvisionerList struct {
 	Items           []Provisioner `json:"items"`
 }
 
-func (p *Provisioner) ConstraintsWithOverrides(pod *v1.Pod) *Constraints {
-	return &Constraints{
-		Taints:          p.Spec.Taints,
-		Labels:          p.Spec.Constraints.getLabels(p.Name, p.Namespace, pod),
-		Zones:           p.Spec.Constraints.getZones(pod),
-		InstanceTypes:   p.Spec.Constraints.getInstanceTypes(pod),
-		Architecture:    p.Spec.Constraints.getArchitecture(pod),
-		OperatingSystem: p.Spec.Constraints.getOperatingSystem(pod),
-	}
+func (c *Constraints) WithLabel(key string, value string) *Constraints {
+	c.Labels = functional.UnionStringMaps(c.Labels, map[string]string{key: value})
```

Comment:
Could this also be done with `c.Labels[key] = value`?  

`functional.UnionStringMaps()` works with "last write wins", so it seems like we overwrite an existing entry in `c.labels` with either versions, so it might be cleaner just to add it in.

Reply:
Oopsie, nice call!